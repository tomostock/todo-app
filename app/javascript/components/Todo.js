import React from "react"
import PropTypes from "prop-types"
import Box from "@material-ui/core/Box";

class Todo extends React.Component {
  render () {
    return (
      <React.Fragment>
        Greetings: {this.props.greeting}
        <p>{ this.props.test }</p>
        <a href={ this.props.path }>new</a>
        <p>{this.props.todo}</p>
      </React.Fragment>
    );
  }
}

//形チェック
Todo.propTypes = {
  greeting: PropTypes.string
};
export default Todo
