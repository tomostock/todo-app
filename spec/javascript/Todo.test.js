/**
 * @jest-environment jsdom
 */
import React from "react"
import { render, unmountComponentAtNode } from "react-dom"
import { act } from "react-dom/test-utils"
import Todo, { NewModal } from "app/javascript/components/Todo"

describe("Todoのレンダー", () => {

  let container = null

  beforeEach(() => {
    container = document.createElement("div")
    document.body.appendChild(container)
  });

  afterEach(() => {
    unmountComponentAtNode(container)
    container.remove()
    container = null
  });

  it("Todo renders propsがある場合", () => {
    const todos = [
      { id: 1, titile: "text1", content: "content1" },
      { id: 2, titile: "text2", content: "content2" } 
    ]
    act(() => {
      render(<Todo todos={ todos }/>, container);
    });
    expect(container.textContent).toMatch(/Delete/);
  })

  it("Todo renders snapshot", () => {
    const todos = [
      { id: 1, titile: "text1", content: "content1" },
      { id: 2, titile: "text2", content: "content2" } 
    ]
    act(() => {
      render(<Todo todos={ todos }/>, container)
    });
    expect(container).toMatchSnapshot()
  })

  it("Todo renders propsがない場合", () => {
    const todos = []
    act(() => {
      render(<Todo todos={ todos }/>, container)
    });
    expect(container.textContent).toBe("TODOnew")
  })
})
