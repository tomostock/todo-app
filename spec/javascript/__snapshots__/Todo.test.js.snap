// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `null`;

exports[`Todoのレンダー Todo renders snapshot 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <h1>
        TODO
      </h1>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text newBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            new
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="todos"
      >
        <div
          class="todos"
        >
          <div
            class="todoContainer"
          >
            <h2 />
            <p>
              content1
            </p>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text detailBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  detail
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text editBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  edit
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text delBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="todoContainer"
          >
            <h2 />
            <p>
              content2
            </p>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text detailBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  detail
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text editBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  edit
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text delBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
